### ISSUE TITLE
Uncaught TypeError: undefined is not a function

### ISSUE THREAD

**Issue by andela-ssunday** [atom-minimap/minimap#371]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 0.175.0
**System**: Mac OS X 10.10.3
**Thrown From**: [minimap](https://github.com/atom-minimap/minimap) package, v4.12.2

### Stack Trace

Uncaught TypeError: undefined is not a function

```
At /Users/andela/.atom/packages/minimap/lib/minimap-element.coffee:89

TypeError: undefined is not a function
  at atom-text-editor-minimap.MinimapElement.attachedCallback (/Users/andela/.atom/packages/minimap/lib/minimap-element.coffee:89:35)
  at atom-pane.PaneElement.activeItemChanged (/Applications/Atom.app/Contents/Resources/app/src/pane-element.js:110:24)
  at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:82:11)
  at Pane.module.exports.Pane.setActiveItem (/Applications/Atom.app/Contents/Resources/app/src/pane.js:264:22)
  at Pane.module.exports.Pane.activateItem (/Applications/Atom.app/Contents/Resources/app/src/pane.js:328:21)
  at Pane.module.exports.Pane.activateItemAtIndex (/Applications/Atom.app/Contents/Resources/app/src/pane.js:322:19)
  at Pane.module.exports.Pane.activateNextItem (/Applications/Atom.app/Contents/Resources/app/src/pane.js:283:21)
  at Pane.module.exports.Pane.removeItem (/Applications/Atom.app/Contents/Resources/app/src/pane.js:391:16)
  at Pane.module.exports.Pane.destroyItem (/Applications/Atom.app/Contents/Resources/app/src/pane.js:457:16)
  at Pane.module.exports.Pane.destroyActiveItem (/Applications/Atom.app/Contents/Resources/app/src/pane.js:436:12)
  at Workspace.module.exports.Workspace.destroyActivePaneItem (/Applications/Atom.app/Contents/Resources/app/src/workspace.js:716:35)
  at Workspace.module.exports.Workspace.destroyActivePaneItemOrEmptyPane (/Applications/Atom.app/Contents/Resources/app/src/workspace.js:760:21)
  at atom-workspace.atom.commands.add.core:close (/Applications/Atom.app/Contents/Resources/app/src/workspace-element.js:292:30)
  at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app/src/command-registry.js:243:29)
  at /Applications/Atom.app/Contents/Resources/app/src/command-registry.js:3:61
  at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/Applications/Atom.app/Contents/Resources/app/node_modules/atom-keymap/lib/keymap-manager.js:558:16)
  at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Applications/Atom.app/Contents/Resources/app/node_modules/atom-keymap/lib/keymap-manager.js:396:22)
  at HTMLDocument.module.exports.WindowEventHandler.onKeydown (/Applications/Atom.app/Contents/Resources/app/src/window-event-handler.js:170:20)

```

### Commands

```
     -0:14.4 core:move-right (atom-pane.pane.active)
     -0:12.9 core:move-left (atom-pane.pane.active)
     -0:08.3 settings-view:open (atom-pane.pane.active)
  2x -0:05.3 fuzzy-finder:toggle-file-finder (div.settings-view.pane-item)
     -0:00.0 core:close (div.settings-view.pane-item)
```

### Config

```json
{
  "core": {
    "themes": [
      "atom-dark-ui",
      "atom-dark-syntax"
    ]
  }
}
```

### Installed Packages

```coffee
# User
color-picker, v2.0.11
minimap, v4.12.2
pretty-json, v0.4.1
vim-mode, v0.57.0

# Dev
No dev packages
```

/cc @atom/core

**Comment by abe33** [comment]
Please update your version of Atom, latest version of the minimap only support Atom >= 0.284 and control of the engine version when updating a package was implemented in Atom 0.176 (hence the error on 0.175).

**Comment by nelson6e65** [comment]
I'm still getting this issue:

**Atom Version**: 1.0.11
**System**: Ubuntu 12.04.4
**Thrown From**: [minimap](https://github.com/atom-minimap/minimap) package, v4.23.1


### Stack Trace

Uncaught TypeError: undefined is not a function

```
At /home/jeahm/.atom/packages/minimap/lib/adapters/legacy-adapter.js:66

TypeError: undefined is not a function
  at LegacyAdater.getMaxScrollTop (/home/jeahm/.atom/packages/minimap/lib/adapters/legacy-adapter.js:90:42)
  at Minimap.getTextEditorMaxScrollTop (/home/jeahm/.atom/packages/minimap/lib/minimap.js:543:27)
  at Minimap.getTextEditorScrollRatio (/home/jeahm/.atom/packages/minimap/lib/minimap.js:606:50)
  at Minimap.getCapedTextEditorScrollRatio (/home/jeahm/.atom/packages/minimap/lib/minimap.js:620:31)
  at Minimap.getScrollTopFromEditor (/home/jeahm/.atom/packages/minimap/lib/minimap.js:992:28)
  at Minimap.getScrollTop (/home/jeahm/.atom/packages/minimap/lib/minimap.js:939:87)
  at Minimap.getFirstVisibleScreenRow (/home/jeahm/.atom/packages/minimap/lib/minimap.js:903:30)
  at atom-text-editor-minimap.canvasLeftMousePressed (/home/jeahm/.atom/packages/minimap/lib/minimap-element.js:1221:82)
  at atom-text-editor-minimap.canvasPressed (/home/jeahm/.atom/packages/minimap/lib/minimap-element.js:1193:14)
  at Object.mousedown [as __NONE__] (/home/jeahm/.atom/packages/minimap/lib/minimap-element.js:506:18)
  at HTMLCanvasElement.listener (/home/jeahm/.atom/packages/minimap/node_modules/atom-utils/lib/mixins/events-delegation.js:108:41)

```

### Commands

```
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "php-getters-setters",
      "line-ending-selector",
      "linter-scss-lint",
      "linter-eslint",
      "php-debug",
      "phpunit",
      "autoclose-html"
    ],
    "packagesWithKeymapsDisabled": []
  },
  "minimap": {}
}
```

### Installed Packages

```coffee
# User
atom-text-transformer, v0.7.0
auto-detect-indentation, v1.0.0
auto-indent, v0.5.0
autocomplete-php, v0.3.7
docblockr, v0.7.3
docs-snippets, v0.8.0
exception-reporting, v0.38.1
html-entities, v0.4.0
indentation-indicator, v0.6.0
language-blade, v0.24.1
language-latte, v0.3.1
line-ending-converter, v1.3.2
linter, v1.11.4
linter-bootlint, v1.0.2
linter-coffeelint, v1.1.2
linter-csslint, v1.3.2
linter-htmlhint, v1.1.4
linter-pep8, v1.2.1
linter-php, v1.2.0
linter-phpcs, v1.3.2
linter-shellcheck, v1.2.0
linter-xmllint, v1.3.4
merge-conflicts, v1.4.2
metrics, v0.53.1
minimap, v4.23.1
pigments, v0.26.0
sync-settings, v0.7.2

# Dev
No dev packages
```

**Comment by abe33** [comment]
@nelson6e65 You are 7 minor Atom versions behind, you should probably update it if you can. 
As for the issue you encountered, I pushed a fix in v4.23.2.
